src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionResolveHandler.cs

```diff
@@ -128,6 +129,16 @@ public async Task<CompletionItem> HandleRequestAsync(CompletionItem request, Cli
             var formattingOptions = _formattingOptionsProvider.GetOptions(requestContext.HostDocumentUri);
             if (resolvedCompletionItem.TextEdit != null)
             {
+                if (!formattingOptions.InsertSpaces)
+                {
+                    // Language servers should technically be formatting these edits with the correct tabs/spaces setting,
+                    // however LSP doesn't currently include formatting options in completion requests. Until they do,
```

Comment:
Definitely an interesting point. @dbaeumer what's the expectation of language servers who need to generate multi-line snippets in their completion items if formatting options are not provided? Aka, does the editor do this type of translation to spaces/tabs based on commit? 

To workaround the limitation that formatting options aren't provided in the completion request I imagine most language servers today try to lookup the last known formatting option (as Allison does above).

Reply:
This can be controlled using the `InsertTextMode` https://github.com/microsoft/vscode-languageserver-node/blob/main/types/src/main.ts#L1967

If `asIs` a server should use the configuration request sent from the server to the client to read the formatting options.

Reply:
> If `asIs` a server should use the configuration request sent from the server to the client to read the formatting options.

Interesting, for embedded scenarios this is something we may need to re-think a bit. We'll be talking about it more in todays sync too so should be a good convo